### DISCUSSION TITLE
Vulnerability and End-of-Life Notification for OpenSSL in NSClient++

### DISCUSSION THREAD

**Discussion by kadhar** [mickem/nscp#836] [General]
Hello Support Team,

We have identified a security vulnerability in our current version of OpenSSL associated with C:\Program Files\NSClient++\ssleay32.dll. The version in use, 1.0.1j, has been End-of-Life (EOL) since December 20, 2019, and lacks support for security patches or vulnerability disclosures.

**Details:**
**File Path**: C:\Program Files\NSClient++\ssleay32.dll
**Current Version**: 1.0.1j (vulnerable to CVE-2021-4044)
**Severity**: High (based on the National Vulnerability Database, CPE cpe:2.3:a:openssl

Since the identified vulnerability affects versions below 1.0.2, we’d like to request guidance on updating OpenSSL to a supported version, ideally 1.0.2 or higher, as a remediation step. Could you advise if there are specific compatibility considerations or steps we need to take to update this component within NSClient++?

Thank you in advance for your support on this matter.

**Comment by mickem** [comment]
I thought openssl was updated as of this year but will investigate and release a new version ASAP if required

**Reply by kadhar** [reply]
Thank you, Mickem. Are there any timelines for the new version release?

**Reply by mickem** [reply]
Which version are you using?
The latest version 0.6.0 https://github.com/mickem/nscp/releases/tag/0.6.0.1 should have version 1.1.1q of openssl

**Reply by kadhar** [reply]
We have installed NSCP versions **0.5.2.35** and **0.5.2.41** on **Windows Server 2016 Datacenter edition**, but we still see an older version of "ssleay32.dll" present.

We attempted to install version **0.6.0** as you recommended, but encountered the following error message:

**"There is a problem with this Windows Installer package. A DLL required for this install to complete could not be run. Contact your support personnel or package vendor."**

Could you provide guidance on how to resolve this installer error, or if there are alternative methods to update the OpenSSL version within NSClient++? 

Any assistance would be greatly appreciated.

**Reply by kadhar** [reply]
![image](https://github.com/user-attachments/assets/e2b70f60-ee55-4181-8e47-1d0fbbb41f72)

**Reply by mickem** [reply]
Can you provide the log file from the installer: `msiexec /l* log.txt /i <MSI FILE>` you can send it to my email at michael@medin.name

**Comment by kadhar** [comment]
Hi Michael,
Please find the logs for NSClient++ (x64) v0.6.0001, where we observed the error, as attached.
[Nsclient_v0.6.0001_Log.txt](https://github.com/user-attachments/files/17773845/Nsclient_v0.6.0001_Log.txt)

**Reply by kadhar** [reply]
I hope you have reviewed the logs. Please let me know your findings so that we can resolve the issue related to the latest NSCP version upgrade and address the vulnerability fix for "ssleay32.dll."

**Reply by mickem** [reply]
Yes, I saw the logs and I am working on a fix, so I will get back once that is live...

**Comment by mickem** [comment]
Sorry for the late reply here is the new version: https://github.com/mickem/nscp/releases/tag/0.6.1.0

Let mw know if this works better, please note that visual studio runtime need to be installed before running thew installer: https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170

**Comment by kadhar** [comment]
Hi Mick, We have applied the new version 0.6.2.1, and the security
vulnerability issue has been fixed.
However, the *NSClient++* service is getting hung up and not responding to
Nagios server, and we have observed the following two events in the Windows
Event Logs. As a result, we are receiving alert notifications for all the
services.
------------------------------
*Event Log Details:*

*Log Name:* System
*Source:* Service Control Manager
*Date:* 1/10/2025 4:51:02 AM
*Event ID:* 7011
*Task Category:* None
*Level:* Error
*Keywords:* Classic
*User:* N/A
*Computer:* awindprodad001.assist-indprod.local

*Description:*
A timeout (30000 milliseconds) was reached while waiting for a transaction
response from the *nscp* service.
[image: image.png]



[image: image.png]

**Reply by mickem** [reply]
Sounds like NSClient++ is not starting correctly then.
Could you provide the log from NSClient++ ?

Should be in the application folder?

// Michael Medin

**Comment by kadhar** [comment]
This is the log we have in the log file


```
025-01-06 07:32:57: error:d:\a\nscp\nscp\modules\NRPEServer\NRPEServer.cpp:121: Certificate not found: C:\Program Files\NSClient++/security/certificate.pem (generating a default certificate)
2025-01-06 07:32:57: error:d:\a\nscp\nscp\modules\NRPEServer\NRPEServer.cpp:121: CA not found: C:\Program Files\NSClient++/security/ca.pem (generating a default CA)
2025-01-06 07:33:13: error:d:\a\nscp\nscp\include\nrpe/server/protocol.hpp:100: Rejected connection from: 10.160.5.100
```

**Reply by mickem** [reply]
Then the problem is that you do not allow connections from: 10.160.5.100

Is this a new install or an upgrade?

If it is an upgrade then I would assume there is some bugs in the upgrader.
if not then it is likely a miss configuration.

Check the config to ensure that the IP above is included in `allowed hosts`:

``ìni
[/settings/default]
allowed hosts = IP
```


Also please note that is a public issue tracker on GitHub so you don't erroneously post IP addresses which you do not want public.

**Reply by mickem** [reply]
A quick follow up.

Seems the installer was not importing old config properly; I have fixed that in https://github.com/mickem/nscp/releases/tag/0.6.4

**Comment by mickem** [comment]
This should be fixed by https://github.com/mickem/nscp/releases/tag/0.6.8
